File: Day 1: Sonar Sweep/part1.cpp
#include <iostream>
#include <vector>
#include <fstream>

using namespace std;

int main(void) {
    // input obtained from https://adventofcode.com/2021/day/1/input
    
    // read from file and insert all integers into vector
    string filename("input.txt");
    vector<int> A;
    int number;

    ifstream input_file(filename);
    if (!input_file.is_open()) {
        cerr << "Could not open the file - '"
             << filename << "'" << endl;
        return EXIT_FAILURE;
    }

    while (input_file >> number) {
        A.push_back(number);
    }

    int count = 0; 
    for (int i = 0; i < A.size(); ++i) {
        while ((i < A.size() - 1) && (A[i] < A[i+1])) {
            count++;
            i++;
        }
    }
    cout<<count<<endl;
    return 0;
}

File: Day 2: Dive!/part1.py
def pos():
    #  input from https://adventofcode.com/2021/day/2/input       
    with open('input.txt', 'r') as f:
        hpos = 0
        depth = 0
        for x in f:
            if x.find('forward') >= 0:
                hpos += int(x[x.find(' ') + 1:])
            elif x.find('up') >= 0:
                depth -= int(x[x.find(' ') + 1:])
            else:
                depth += int(x[x.find(' ') + 1:])

    print(hpos*depth)

if __name__ == "__main__":
    pos()
    pass
